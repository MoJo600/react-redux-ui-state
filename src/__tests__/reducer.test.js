import { generateName, uiStateReducer } from '../';

jest.mock('uuid/v4', () => () => '110ec58a-a0f2-4ac4-8393-c866d813b8d1');

describe('uiStateReducer', () => {
    const name = generateName('action-test-component');
    const initialState = {
        [name]: {},
    };

    it('should add component intitial state', () => {
        const addAction = {
            payload: {
                name,
                state: {
                    val1: false,
                    val2: true,
                },
            },
            type: `UI_STATE_ADD:${name}`,
        };

        const state = uiStateReducer(initialState, addAction);

        expect(state).toMatchSnapshot();
    });

    it('should update component state', () => {
        const setAction = {
            payload: {
                name,
                state: {
                    val1: true,
                    val2: false,
                    val3: 'what',
                },
            },
            type: `UI_STATE_SET:${name}`,
        };

        const state = uiStateReducer({}, setAction);

        expect(state).toMatchSnapshot();
    });
});
