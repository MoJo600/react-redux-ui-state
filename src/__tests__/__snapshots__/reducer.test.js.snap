// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiStateReducer should add component intitial state 1`] = `
Object {
  "action-test-component_110ec58a-a0f2-4ac4-8393-c866d813b8d1": Object {
    "val1": false,
    "val2": true,
  },
}
`;

exports[`uiStateReducer should update component state 1`] = `
Object {
  "val1": true,
  "val2": false,
  "val3": "what",
}
`;
